LanguageFeatures/Declaring-constructors/static_processing_A24_t01.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The semantics of the declaring constructor is found in the
+/// following steps, where `D` is the class, extension type, or enum declaration
+/// in the program that includes a declaring constructor `k`, and `D2` is the
+/// result of the derivation of the semantics of `D`. The derivation step will
+/// delete elements that amount to the declaring constructor. Semantically, it
+/// will add a new constructor `k2`, and it will add zero or more instance
+/// variable declarations.
+/// ...
+/// If there is an initializer list following the formal parameter list `L` then
+/// `k2` has an initializer list with the same elements in the same order.
+///
+/// @description Check that if a declaring constructor has an initializer list
+/// then `k2` has an initializer list with the same elements.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=declaring-constructors
+
+import '../../Utils/expect.dart';
+
+class A(int z);
+
+class C1 extends A {
+  int x;
+  this: x = 1, super(-1);
```

Comment:
It's an error to use `this:` here unless there is a primary constructor. To make it an in-body declaring constructor it needs to have a formal parameter list, e.g., `()`.

Reply:
Ah, yes. But the case with an empty primary constructor `()` is `C2`. Removed this class.